### ISSUE TITLE
[import/no-cycle] only type imports

### ISSUE THREAD

**Issue by ValeryVS** [typescript-eslint/typescript-eslint#843]
**Repro**

```js
{
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json',
  },
  plugins: [
    '@typescript-eslint',
    'react-hooks',
  ],
  extends: [
    'airbnb',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
  ],
  // ...
}
```

```TS
// store.ts
import orders from 'entities/orders/reducers';

const rootReducer = combineReducers({
  orders,
});

export type AppState = ReturnType<typeof rootReducer>
```

```TS
// entities/orders/reducers.ts
import {SetOrdersAction} from './actions';

export default createReducer(initialState, {
 ...
})
```

```TS
// entities/orders/actions.ts
import {ThunkAction} from 'redux-thunk';

import {AppState} from 'store';

export const pollOrders = (): ThunkAction<void, AppState, void, SetOrdersAction> => (
   ...
);
```

**Actual Result**

There is the store (imports reducers), reducers (imports actions), and actions (imports store type) in application.
`import/no-cycle` rule alerts about cycle dependency, but only type information are cycled and it will be skipped at runtime

Is it safe to ignore such cycles now?
May be it will be good to make typescript version of this rule and ignore types in cycle detecting?

Or, may be there is better way to declare `AppState` type?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Hi there @ValeryVS - we don't have any direct association with the `import` plugin.
You should raise this issue on their repo so you can discuss with that project's maintainers.
